straxen/plugins/event_processing.py

```diff
@@ -522,27 +522,32 @@ class EventPositions(strax.Plugin):
                 '?version=ONLINE&run_id=plugin.run_id',
         help='Electron drift time from the gate in ns',
         cache=True)
-
-    dtype = [
-        ('x', np.float32,
-         'Interaction x-position, field-distortion corrected (cm)'),
-        ('y', np.float32,
-         'Interaction y-position, field-distortion corrected (cm)'),
-        ('z', np.float32,
-         'Interaction z-position, using mean drift velocity only (cm)'),
-        ('r', np.float32,
-         'Interaction radial position, field-distortion corrected (cm)'),
-        ('z_naive', np.float32,
-         'Interaction z-position using mean drift velocity only (cm)'),
-        ('r_naive', np.float32,
-         'Interaction r-position using observed S2 positions directly (cm)'),
-        ('r_field_distortion_correction', np.float32,
-         'Correction added to r_naive for field distortion (cm)'),
-        ('z_field_distortion_correction', np.float32,
-         'Correction added to z_naive for field distortion (cm)'),
-        ('theta', np.float32,
-         'Interaction angular position (radians)')
-            ] + strax.time_fields
+    
+    dtype = []
+    for j in 'x y r z'.split():
+        comment = f'Main interaction {j}-position, field-distortion corrected (cm)' if j != 'z' \
+                  else 'Interaction z-position, using mean drift velocity only (cm)'
+        dtype += [(j, np.float32, comment)]
+        for s_i in [1, 2]:
+            comment = f'Alternative S{s_i} interaction {j}-position, field-distortion corrected (cm)' if j != 'z' \
+                      else f'Alternative S{s_i} z-position, using mean drift velocity only (cm)'
+            dtype += [(f'alt_s{s_i}_{j}_fdc' if j != 'z' else f'alt_s{s_i}_z', np.float32, comment)]
+    
+    for j in 'r z'.split():
+        dtype += [('_'.join([j, 'naive']), np.float32, f'Main interaction {j}-position with observed position (cm)')]
+        for s_i in [1, 2]:
+            dtype += [(f'alt_s{s_i}_{j}_naive', np.float32, f'Alternative S{s_i} interaction {j}-position with observed position (cm)')]
+            
+    for j in 'r z'.split():
+        dtype += [('_'.join([j, 'field_distortion_correction']), np.float32, f'Correction added to {j}_naive for field distortion (cm)')]
+        for s_i in [1, 2]:
+            dtype += [(f'alt_s{s_i}_{j}_field_distortion_correction', np.float32, f'Correction added to alt_s{s_i}_{j}_naive for field distortion (cm)')]
+            
+    dtype += [('theta', np.float32, f'Main interaction angular position (radians)')]
+    for s_i in [1, 2]:
+        dtype += [(f'alt_s{s_i}_theta', np.float32, f'Alternative S{s_i} interaction angular position (radians)')]
+    
+    dtype = dtype + strax.time_fields
```

Comment:
```suggestion
def infer_dtype(self):
        dtype = []
        for j in 'x y r'.split():
            comment = f'Main interaction {j}-position, field-distortion corrected (cm)'
            dtype += [(j, np.float32, comment)]
            for s_i in [1, 2]:
                comment = f'Alternative S{s_i} interaction {j}-position, field-distortion corrected (cm)'
                field = f'alt_s{s_i}_{j}_fdc'
                dtype += [(field, np.float32, comment)]

        for j in ['z']:
            comment = 'Interaction z-position, using mean drift velocity only (cm)'
            dtype += [(j, np.float32, comment)]
            for s_i in [1, 2]:
                comment = f'Alternative S{s_i} z-position, using mean drift velocity only (cm)'
                field = f'alt_s{s_i}_z'
                dtype += [(field, np.float32, comment)]

        naive_pos = []
        fdc_pos = []
        for j in 'r z'.split():
            naive_pos += [(f'{j}_naive',
                           np.float32,
                           f'Main interaction {j}-position with observed position (cm)')]
            fdc_pos += [(f'{j}_field_distortion_correction',
                         np.float32,
                         f'Correction added to {j}_naive for field distortion (cm)')]
            for s_i in [1, 2]:
                naive_pos += [(
                    f'alt_s{s_i}_{j}_naive',
                    np.float32,
                    f'Alternative S{s_i} interaction {j}-position with observed position (cm)')]
                fdc_pos += [(f'alt_s{s_i}_{j}_field_distortion_correction',
                             np.float32,
                             f'Correction added to alt_s{s_i}_{j}_naive for field distortion (cm)')]
        dtype += naive_pos + fdc_pos
        for s_i in [1, 2]:
            dtype += [(f'alt_s{s_i}_theta',
                       np.float32,
                       f'Alternative S{s_i} interaction angular position (radians)')]

        dtype += [('theta', np.float32, f'Main interaction angular position (radians)')]
        return dtype + strax.time_fields
```
I had a hard time following this, perhaps something like this makes it a bit easier to read?

Reply:
I think this is nicer,  I am thinking that there are some fields that do not really make sense (e.g., `alt_s1_theta` and `alt_s1_r_naive` are identical to `theta` or `r_naive`). Would it be better not to include them or is it worth to have more space but having more homogeneity in the field names?